src/Domain/Entities/OrderItem.cs

```diff
@@ -1,12 +1,22 @@
 ﻿using Domain.Common;
+using System.ComponentModel.DataAnnotations;
 
 namespace Domain.Entities
 {
     public class OrderItem : BaseEntity
     {
+        [Required]
         public int OrderId { get; set; }
+
+        [Required]
         public int ProductId { get; set; }
+
+        [Required]
         public int Quantity { get; set; }
+
+        [Required]
+        [RegularExpression(@"^\d+\.\d{0,2}$")]
+        [Range(0, 9999999999999999.99, ErrorMessage = "Invalid Price; Max 18 digits")]
```

Comment:
Both RegExp and Range is not necessary. Use RegExp `^\d{1,18}(\.\d{1,2}){0,1}$` or the Range.